sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobCheckpointStoreInternal.cs

```diff
@@ -437,16 +438,26 @@ await foreach (BlobItem blob in ContainerClient.GetBlobsAsync(traits: BlobTraits
             var startingPosition = default(EventPosition?);
             var offset = default(long?);
             var sequenceNumber = default(long?);
+            var clientIdentifier = default(string);
+            var replicationSegment = default(string);
 
-            if (metadata.TryGetValue(BlobMetadataKey.Offset, out var str) && long.TryParse(str, NumberStyles.Integer, CultureInfo.InvariantCulture, out var result))
+            if (metadata.TryGetValue(BlobMetadataKey.ReplicationSegment, out var str))
```

Comment:
Should we allow this to be set if there is no value for sequence number?

Reply:
It won't get used if there's no sequence number, so `null` seems fine.